src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -138,6 +138,29 @@ public function dump(array $options = array())
 
         $this->initializeMethodNamesMap('Container' === $baseClass ? Container::class : $baseClass);
 
+        if ($this->getProxyDumper() instanceof NullDumper) {
+            (new AnalyzeServiceReferencesPass(true))->process($this->container);
+            $this->circularReferences = array();
+            $checkedNodes = array();
+            foreach ($this->container->getCompiler()->getServiceReferenceGraph()->getNodes() as $id => $node) {
+                $currentPath = array($id => $id);
+                $this->analyzeCircularReferences($node->getOutEdges(), $checkedNodes, $currentPath);
+            }
+            foreach ($this->circularReferences as $parent => $ids) {
+                $path = array($parent);
+                while (!isset($ids[$parent])) {
+                    foreach ($ids as $id) {
+                        $path[] = $id;
+                        $ids = $this->circularReferences[$id];
+                        break;
+                    }
+                }
+                $path[] = $parent.'". Try running "composer require symfony/proxy-manager-bridge';
+
+                throw new ServiceCircularReferenceException($parent, $path);
```

Comment:
Don't dump broken service factories when lazy services are not really lazy because proxy-manager is not here. Let's throw an exception suggesting to install it instead.